videostore/videostore.go

```diff
@@ -339,27 +340,32 @@ func (vs *videostore) fetchFrames(ctx context.Context, cam camera.Camera) {
 				time.Sleep(retryInterval * time.Second)
 				continue
 			}
-			var frame *C.AVFrame
-
-			mimetype, _ := rutils.CheckLazyMIMEType(metadata.MimeType)
-			switch mimetype {
-			case mimeTypeYUYV:
-				frame, err = mimeHandler.yuyvToYUV420p(bytes)
-				if err != nil {
-					vs.logger.Error("failed to convert yuyv422 to yuv420p", err)
-					continue
-				}
-			case rutils.MimeTypeJPEG, rutils.MimeTypeJPEG + "+" + rutils.MimeTypeSuffixLazy:
-				frame, err = mimeHandler.decodeJPEG(bytes)
-				if err != nil {
-					vs.logger.Error("failed to decode jpeg", err)
-					continue
+			// Transform image bytes to yuv420p based on the mimetype.
+			func() {
```

Comment:
Wrapping in a function so we can always defer unlocking making it simpler to handle error cases.

Reply:
Add the reasoning to the comment above.